src/hooks/useUrlBar.ts

```diff
@@ -0,0 +1,87 @@
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import { useCallback, useMemo } from "react";
+
+export function useUrlBar() {
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const updateSearchParam = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams?.toString());
+      params.set(name, value);
+      router.push(`${pathname}?${params.toString()}`);
+    },
+    [pathname, router, searchParams],
+  );
+
+  const addSearchParam = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams?.toString());
+      params.append(name, value);
+      router.push(`${pathname}?${params.toString()}`);
+    },
+    [pathname, router, searchParams],
+  );
+
+  const addSearchParams = useCallback(
+    (params: Record<string, string>) => {
+      const urlSearchParams = new URLSearchParams(searchParams?.toString());
+
+      for (const [key, value] of Object.entries(params)) {
+        urlSearchParams.set(key, value);
+      }
+
+      router.push(`${pathname}?${urlSearchParams.toString()}`);
+    },
+    [pathname, router, searchParams],
+  );
+
+  const removeSearchParam = useCallback(
```

Comment:
super annoyingly, this is the only way to remove a param with a given name and value.

the api is supposed to support doing this https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/delete

but its only available in firefox at the moment.so we need this workaround